x/programs/cmd/simulator/src/lib.rs

```diff
@@ -198,7 +169,72 @@ pub struct PlanResult {
     /// The timestamp of the function call response.
     pub timestamp: u64,
     /// The result of the function call.
-    pub response: Option<String>,
+    #[serde_as(as = "Base64")]
+    pub response: Vec<u8>,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct BaseResultTyped<T>
+where
+    T: BorshDeserialize,
+{
+    /// The ID created from the program execution.
+    pub id: Option<String>,
+    /// An optional message.
+    pub msg: Option<String>,
+    /// The timestamp of the function call response.
+    pub timestamp: u64,
+    /// The result of the function call.
+    pub response: T,
+}
```

Comment:
I was actually not trivial to use a serde `tag` here to define an enum with a typed and an untyped variant. The reason is that we will have to deal with the variants in the `TryInto` function.